src/requirements.cpp

```diff
@@ -31,6 +31,17 @@ void quality::load_static( JsonObject &jo )
 void quality::load( JsonObject &jo )
 {
     mandatory( jo, was_loaded, "name", name, translated_string_reader );
+
+    JsonArray arr = jo.get_array( "usages" );
+    while( arr.has_more() ) {
+        auto lvl = arr.next_array();
+        if( lvl.has_int( 0 ) && lvl.has_array( 1 ) ) {
```

Comment:
Why test this at all? Just access it and let the Json classes throw an error if the data is there / not in the correct format. This code is a bit unfriendly to the user, they make one mistake (e.g. switching the order: string first, number second) and the game just *ignores* the entry. No error message or anything.